lib/services/role.go

```diff
@@ -2394,10 +2408,10 @@ func (p boolParser) Parse(string) (interface{}, error) {
 // CheckAccessToRule checks if the RoleSet provides access in the given
 // namespace to the specified resource and verb.
 // silent controls whether the access violations are logged.
-func (set RoleSet) CheckAccessToRule(ctx RuleContext, namespace string, resource string, verb string, silent bool) error {
+func (set RoleSet) CheckAccessToRule(ctx RuleContext, namespace string, resource string, verb string, silent bool) (predicateEngine.AccessDecision, error) {
 	whereParser, err := NewWhereParser(ctx)
 	if err != nil {
-		return trace.Wrap(err)
+		return predicateEngine.AccessUndecided, trace.Wrap(err)
```

Comment:
why undecided when there is a parse error? should we play it safe and deny access unless we know for sure it is meant to be undecided?

Reply:
Throughout this code, undecided has been used as a sort of default value. This still errors, though, so it'll never return allow and no error at the top level. I think undecided better reflects how we'd reason about this mentally but it largely doesn't matter functionally here as the error ensures you never get that higher level access decision.

Reply:
ah, somehow I didn't consider the error return, that's fine then